File: haskell/fizzBuzz.hs
-- FizzBuzz in Haskell
main :: IO()
main = do
    numTimesStr <- (prompt "FizzBuzz: Enter an integer: ")
    let numTimes = read numTimesStr :: Int
    putStrLn (show [fbEval x | x <- [1..numTimes]])
    main

prompt :: String -> IO String
prompt x = do
    putStr x
    getLine

fbEval :: Int -> String
fbEval x
    | x `rem` 15 == 0 = "FizzBuzz"
    | x `rem` 3 == 0= "Fizz"
    | x `rem` 5 == 0 = "Buzz"
    | otherwise = show x

File: c/FizzBuzz.c
// FizzBuzz in C
// Compile with "gcc FizzBuzz.c"

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void fizzBuzz(int min, int max) {
    for (int current = min; current <= max; ++current) {
        if (current % 3 == 0) {
            printf("Fizz");
        }
        if (current % 5 == 0) {
            printf("Buzz");
        }
        if (current % 3 != 0 && current % 5 != 0) {
            printf("%d", current);
        }

        printf("\n");
    }
}

char* getLine() {
    size_t lenMax = 100;
    size_t len = lenMax;
    char *line = malloc(sizeof(char) * len);
    char *linep = line;

    int c;

    if (line == NULL) return NULL;

    for (;;) {
        c = fgetc(stdin);
        if (c == EOF) break;

        if (--len == 0) {
            len = lenMax;
            char *lineNew = realloc(linep, sizeof(char) * (lenMax *= 2));

            if (lineNew == NULL) {
                free(linep);
                return NULL;
            }

            line = lineNew + (line - linep);
            linep = lineNew;
        }

        if ((*line++ = c) == '\n') break;
    }
    *line = '\0';
    return linep;
}

int main() {
    printf("FizzBuzz: Enter an integer > ");
    char *numTimesStr = getLine();
    int numTimes = atoi(numTimesStr);

    fizzBuzz(1, numTimes);

    return 0;
}
